README.md
# [Soft UI Flowbite](http://demos.creative-tim.com/soft-ui-flowbite/?ref=readme-suf) [![Tweet](https://img.shields.io/twitter/url/http/shields.io.svg?style=social&logo=twitter)](https://twitter.com/intent/tweet?url=https://www.creative-tim.com/product/soft-ui-flowbite&text=Check%20Soft%20UI%20Flowbite%20made%20by%20@CreativeTim%20#webdesign%20#dashboard%20#flowbite%20#softdesign%20#react%20https://www.creative-tim.com/product/soft-ui-flowbite)

![version](https://img.shields.io/badge/version-1.0.0-blue.svg) [![GitHub issues open](https://img.shields.io/github/issues/creativetimofficial/soft-ui-flowbite.svg)](https://github.com/creativetimofficial/soft-ui-flowbite/issues?q=is%3Aopen+is%3Aissue) [![GitHub issues closed](https://img.shields.io/github/issues-closed-raw/creativetimofficial/soft-ui-flowbite.svg)](https://github.com/creativetimofficial/soft-ui-flowbite/issues?q=is%3Aissue+is%3Aclosed)

![Image](https://s3.amazonaws.com/creativetim_bucket/products/644/original/soft-ui-flowbite.jpg)

Build your website faster with an Innovative Admin Template based on TailwindCSS, Flowbite, and HTML. Soft UI Flowbite features a huge number of components built to fit together and look amazing.

If you like the look & feel of the hottest design trends right now, you will definitely fall in love with this dashboard. Its design is inspired by Soft UI Design and it gives you the flexibility that will get you going in no time.

If you are looking to take your app to the next level – Soft UI Flowbite is the way to go.

**Fully Coded Components**<br />
Soft UI Flowbite is built with over 30 frontend individual elements, like buttons, inputs, navbars, nav tabs, cards, or alerts, giving you the freedom of choosing and combining. All components can take variations in color, which you can easily modify using TailwindCSS config and classes. You will save a lot of time going from prototyping to full-functional code because all elements are implemented.

This Free Dashboard is coming with prebuilt design blocks, so the development process is seamless, switching from our pages to the real website is very easy to be done.

View [all components here](https://www.creative-tim.com/learning-lab/flowbite/alerts/soft-ui-dashboard/).

**Documentation built by Developers**<br />
Each element is well presented in very complex documentation.
You can read more about the [documentation here](https://www.creative-tim.com/learning-lab/flowbite/overview/soft-ui-dashboard/).

**Example Pages**<br />
If you want to get inspiration or just show something directly to your clients, you can jump-start your development with our 5 prebuilt example pages. You will be able to quickly set up the basic structure for your web project.

View [example pages here](https://demos.creative-tim.com/soft-ui-flowbite/).

**HELPFUL LINKS**

- View [Github Repository](https://github.com/creativetimofficial/soft-ui-flowbite)
- Check [FAQ Page](https://www.creative-tim.com/faq)

#### Special thanks

During the development of this dashboard, we have used many existing resources from awesome developers. We want to thank them for providing their tools open source:

- [Flowbite](https://flowbite.com/) - TailwindCSS components library.
- [Apex Charts](https://apexcharts.com/) - Modern & Interactive Open-source Charts.

Let us know your thoughts below. And good luck with development!

## Table of Contents

- [Versions](#versions)
- [Demo](#demo)
- [Quick Start](#quick-start)
- [Documentation](#documentation)
- [File Structure](#file-structure)
- [Browser Support](#browser-support)
- [Resources](#resources)
- [Reporting Issues](#reporting-issues)
- [Technical Support or Questions](#technical-support-or-questions)
- [Licensing](#licensing)
- [Useful Links](#useful-links)

## Versions

[<img src="https://raw.githubusercontent.com/creativetimofficial/public-assets/master/logos/html-logo.jpg?raw=true" width="60" height="60" />](https://www.creative-tim.com/product/soft-ui-flowbite-pro?ref=readme-suf)

| HTML |
| ---- |

| [![Soft UI Flowbite](https://s3.amazonaws.com/creativetim_bucket/products/644/thumb/soft-ui-flowbite.jpg)](http://demos.creative-tim.com/soft-ui-flowbite/?ref=readme-suf)

## Demo

- [Dashboard](http://demos.creative-tim.com/soft-ui-flowbite/?ref=readme-suf)
- [Sign In](https://demos.creative-tim.com/soft-ui-flowbite/sign-in?ref=readme-suf)
- [Sign Up](https://demos.creative-tim.com/soft-ui-flowbite/sign-up?ref=readme-suf)

[View More](https://demos.creative-tim.com/soft-ui-flowbite/?ref=readme-suf).

## Quick start

Quick start options:

- Download from [Creative Tim](https://www.creative-tim.com/product/soft-ui-flowbite?ref=readme-suf).

## Terminal Commands

1. Download and Install NodeJs LTS version from [NodeJs Official Page](https://nodejs.org/en/download/).
2. Navigate to the root ./ directory of the product and run `yarn install` or `npm install` to install our local dependencies.

## Documentation

The documentation for the Soft UI Flowbite is hosted at our [website](https://www.creative-tim.com/learning-lab/flowbite/overview/soft-ui-dashboard/?ref=readme-suf).

### What's included

Within the download you'll find the following directories and files:

```
soft-ui-flowbite
    ├── content
    │   ├── _index.html
    │   ├── products.html
    │   ├── sign-in.html
    │   ├── sign-up.html
    │   └── users.html
    ├── data
    ├── layouts
    │   ├── _default
    │   ├── partials
    │   ├── shortcodes
    │   ├── alias.html
    │   └── sitemap.xml
    ├── resources
    ├── src
    ├── static
    ├── .gitignore
    ├── .hugo_build.lock
    ├── CHANGELOG.md
    ├── config.yml
    ├── ISSUE_TEMPLATE.md
    ├── LICENSE
    ├── package.json
    ├── postcss.config.json
    ├── README.md
    ├── syntax.css
    ├── tailwind.config.js
    └── webpack.config.js
```

## Browser Support

At present, we officially aim to support the last two versions of the following browsers:

<img src="https://s3.amazonaws.com/creativetim_bucket/github/browser/chrome.png" width="64" height="64"> <img src="https://s3.amazonaws.com/creativetim_bucket/github/browser/firefox.png" width="64" height="64"> <img src="https://s3.amazonaws.com/creativetim_bucket/github/browser/edge.png" width="64" height="64"> <img src="https://s3.amazonaws.com/creativetim_bucket/github/browser/safari.png" width="64" height="64"> <img src="https://s3.amazonaws.com/creativetim_bucket/github/browser/opera.png" width="64" height="64">

## Resources

- [Live Preview](https://demos.creative-tim.com/soft-ui-flowbite-pro/?ref=readme-suf)
- [Buy Page](https://www.creative-tim.com/product/soft-ui-flowbite-pro?ref=readme-suf)
- Documentation is [here](https://www.creative-tim.com/learning-lab/flowbite/overview/soft-ui-dashboard/?ref=readme-suf)
- [License Agreement](https://www.creative-tim.com/license?ref=readme-suf)
- [Support](https://www.creative-tim.com/contact-us?ref=readme-suf)
- Issues: [Github Issues Page](https://github.com/creativetimofficial/ct-soft-ui-flowbite-pro/issues)

## Reporting Issues

We use GitHub Issues as the official bug tracker for the Soft UI Flowbite PRO. Here are some advices for our users that want to report an issue:

1. Make sure that you are using the latest version of the Soft UI Flowbite PRO. Check the CHANGELOG from your dashboard on our [website](https://www.creative-tim.com/product/soft-ui-flowbite-pro?ref=readme-suf).
2. Providing us reproducible steps for the issue will shorten the time it takes for it to be fixed.
3. Some issues may be browser specific, so specifying in what browser you encountered the issue might help.

## Technical Support or Questions

If you have questions or need help integrating the product please [contact us](https://www.creative-tim.com/contact-us?ref=readme-suf) instead of opening an issue.

## Licensing

- Copyright 2022 [Creative Tim](https://www.creative-tim.com?ref=readme-suf)
- Creative Tim [license](https://www.creative-tim.com/license?ref=readme-suf)

## Useful Links

- [More products](https://www.creative-tim.com/templates?ref=readme-suf) from Creative Tim

- [Tutorials](https://www.youtube.com/channel/UCVyTG4sCw-rOvB9oHkzZD1w)

- [Freebies](https://www.creative-tim.com/bootstrap-themes/free?ref=readme-suf) from Creative Tim

- [Affiliate Program](https://www.creative-tim.com/affiliates/new?ref=readme-suf) (earn money)

##### Social Media

Twitter: <https://twitter.com/CreativeTim>

Facebook: <https://www.facebook.com/CreativeTim>

Dribbble: <https://dribbble.com/creativetim>

Google+: <https://plus.google.com/+CreativetimPage>

Instagram: <https://instagram.com/creativetimofficial>


CHANGELOG.md
# Change Log

## [1.0.0] 2022-06-17

### Initial Release


src/sidebar.js
let toggleAccess = true;

window.addEventListener("resize", () => {
  toggleAccess = window.innerWidth > 1024 ? true : false;
});
window.addEventListener("load", () => {
  toggleAccess = window.innerWidth > 1024 ? true : false;
});

const sidebar = document.getElementById("sidebar");
const sidebarItems = document.getElementById("sidebar-items");

document
  .querySelectorAll("#" + sidebar.getAttribute("id") + " ul > li > ul > li > a")
  .forEach((e) => {
    var fullText = e.textContent;
    var firstLetter = fullText.substring(0, 1);

    var fullTextEl = document.createElement("span");
    var firstLetterEl = document.createElement("span");
    firstLetterEl.classList.add("hidden");
    fullTextEl.textContent = fullText;
    firstLetterEl.textContent = firstLetter;

    e.textContent = "";
    e.appendChild(fullTextEl);
    e.appendChild(firstLetterEl);
  });

const toggleSidebarMobile = (
  sidebar,
  sidebarBackdrop,
  toggleSidebarMobileHamburger,
  toggleSidebarMobileClose
) => {
  sidebar.classList.toggle("hidden");
  sidebarBackdrop.classList.toggle("hidden");
  toggleSidebarMobileHamburger.classList.toggle("hidden");
  toggleSidebarMobileClose.classList.toggle("hidden");
};

const toggleSidebarMobileEl = document.getElementById("toggleSidebarMobile");
const sidebarBackdrop = document.getElementById("sidebarBackdrop");
const toggleSidebarMobileHamburger = document.getElementById(
  "toggleSidebarMobileHamburger"
);
const toggleSidebarMobileClose = document.getElementById(
  "toggleSidebarMobileClose"
);
const toggleSidebarMobileSearch = document.getElementById(
  "toggleSidebarMobileSearch"
);

toggleSidebarMobileSearch.addEventListener("click", () => {
  toggleSidebarMobile(
    sidebar,
    sidebarBackdrop,
    toggleSidebarMobileHamburger,
    toggleSidebarMobileClose
  );
});

toggleSidebarMobileEl.addEventListener("click", () => {
  toggleSidebarMobile(
    sidebar,
    sidebarBackdrop,
    toggleSidebarMobileHamburger,
    toggleSidebarMobileClose
  );
});

sidebarBackdrop.addEventListener("click", () => {
  toggleSidebarMobile(
    sidebar,
    sidebarBackdrop,
    toggleSidebarMobileHamburger,
    toggleSidebarMobileClose
  );
});


tailwind.config.js
module.exports = {
  content: [
    "./layouts/**/*.html",
    "./content/**/*.md",
    "./content/**/*.html",
    "./src/*.js",
    "./node_modules/flowbite/**/*.js",
    "./node_modules/sortablejs/**/*.js"
  ],
  safelist: [
    "w-64",
    "w-1/2",
    "rounded-l-lg",
    "rounded-r-lg",
    "bg-gray-200",
    "grid-cols-4",
    "grid-cols-7",
    "h-6",
    "leading-6",
    "h-9",
    "leading-9",
    "shadow-lg",
    "bg-opacity-50"
  ],
  theme: {
    extend: {
      fontFamily: {
        sans: ["Open Sans", "sans-serif"],
        body: ["Open Sans", "sans-serif"],
        mono: [
          "ui-monospace",
          "SFMono-Regular",
          "Menlo",
          "Monaco",
          "Consolas",
          "Liberation Mono",
          "Courier New",
          "monospace"
        ]
      },
      boxShadow: {
        sm: "0 2px 9px -5pc rgb(0 0 0 / 0.15)",
        DEFAULT: "0 5px 10px 0 rgb(0 0 0 / 0.12)",
        md: "0 4px 6px -1px rgb(0 0 0 / 0.12), 0 2px 4px -1px rgb(0 0 0 / 0.07)",
        lg: "0 8px 26px -4px rgb(0 0 0 / 0.15), 0 8px 9px -5px rgb(0 0 0 / 0.06)",
        xl: "0 23px 45px -11px rgb(0 0 0 / 0.25)",
        "2xl": "0 20px 27px 0 rgb(0 0 0 / 0.05)",
        inner: "inset 0 1px 2px 0 rgb(0 0 0 / 0.075)",
        none: "0 0 #000"
      },
      transitionProperty: {
        width: "width"
      },
      minWidth: {
        20: "20rem"
      },
      colors: {
        fuchsia: {
          50: "#fbb2ea",
          100: "#f88bdf",
          200: "#f665d4",
          300: "#f43ec9",
          400: "#f218bf",
          500: "#cb0c9f",
          600: "#c10b96",
          700: "#ae0a87",
          800: "#9a0978",
          900: "#870869"
        },
        voilet: {
          50: "#e0ccf5",
          100: "#ccaaee",
          200: "#b888e7",
          300: "#a366e1",
          400: "#8f44da",
          500: "#7928CA",
          600: "#7025bb",
          700: "#6622aa",
          800: "#5c1e99",
          900: "#521b88"
        },
        pink: {
          50: "#ffd6eb",
          100: "#ffadd6",
          200: "#ff85c2",
          300: "#ff5cad",
          400: "#ff3399",
          500: "#ff0080",
          600: "#f5007a",
          700: "#e00070",
          800: "#cc0066",
          900: "#b8005c"
        },
        slate: {
          50: "#c2c9d6",
          100: "#b6b7cd",
          200: "#aab4c5",
          300: "#a8b8d8",
          400: "#96a2b8",
          500: "#8392ab",
          600: "#627594",
          700: "#586b86",
          800: "#536179",
          900: "#4a576d"
        },
        gray: {
          50: "#f9fafb",
          100: "#f8f9fa",
          200: "#e9ecef",
          300: "#dee2e6",
          400: "#ced4da",
          500: "#adb5bd",
          600: "#6c757d",
          700: "#495057",
          800: "#343a40",
          900: "#212529"
        },
        cyan: {
          50: "#b5ebf8",
          100: "#90e2f4",
          200: "#69d8f0",
          300: "#21d4fd",
          400: "#3acaeb",
          500: "#17c1e8",
          600: "#16badf",
          700: "#14aacc",
          800: "#129bba",
          900: "#118ba7"
        },
        blue: {
          50: "#859fff",
          100: "#708fff",
          200: "#5c7fff",
          300: "#476fff",
          400: "#335fff",
          500: "#2152ff",
          600: "#0a3fff",
          700: "#0035f5",
          800: "#0031e0",
          900: "#002ccc"
        },
        green: {
          50: "#b6f16a",
          100: "#a4ef57",
          200: "#a4ee44",
          300: "#98ec2d",
          400: "#95dc39",
          500: "#82d616",
          600: "#7ccb15",
          700: "#17ad37",
          800: "#15a234",
          900: "#13902e"
        },
        yellow: {
          50: "#fde387",
          100: "#fcde73",
          200: "#fcda57",
          300: "#fcd652",
          400: "#fbd54b",
          500: "#fbcf33",
          600: "#f8cc23",
          700: "#fac70f",
          800: "#f0bd05",
          900: "#dcad04"
        },
        red: {
          50: "#f98b8b",
          100: "#f87777",
          200: "#f76464",
          300: "#f65151",
          400: "#f53939",
          500: "#ea0606",
          600: "#c70505",
          700: "#b30505",
          800: "#9f0404",
          900: "#8b0404"
        },
        dark: {
          50: "#92a6c8",
          100: "#859bc1",
          200: "#6a85b4",
          300: "#5270a3",
          400: "#445d88",
          500: "#344767",
          600: "#3a416f",
          700: "#383f6b",
          800: "#31375e",
          900: "#141727"
        }
      }
    }
  },

  plugins: [require("flowbite/plugin")]
};


src/charts.js
import ApexCharts from "apexcharts";

if (document.getElementById("sales-overview")) {
  const options = {
    chart: {
      height: 420,
      type: "area",
      fontFamily: "Open Sans, sans-serif",
      foreColor: "#6B7280",
      toolbar: {
        show: false
      }
    },
    fill: {
      type: "gradient",
      gradient: {
        enabled: true,
        opacityFrom: 0.45,
        opacityTo: 0
      }
    },
    dataLabels: {
      enabled: false
    },
    tooltip: {
      style: {
        fontSize: "14px",
        fontFamily: "Open Sans, sans-serif"
      }
    },
    grid: {
      show: true,
      borderColor: "#F3F4F6",
      strokeDashArray: 1,
      padding: {
        left: 35,
        bottom: 15
      }
    },
    series: [
      {
        name: "Revenue",
        data: [6356, 6218, 6156, 6526, 6356, 6256, 6056],
        color: "#0694a2"
      },
      {
        name: "Revenue (previous period)",
        data: [6556, 6725, 6424, 6356, 6586, 6756, 6616],
        color: "#4E4187"
      }
    ],
    markers: {
      size: 5,
      strokeColors: "#ffffff",
      hover: {
        size: undefined,
        sizeOffset: 3
      }
    },
    xaxis: {
      categories: [
        "01 Feb",
        "02 Feb",
        "03 Feb",
        "04 Feb",
        "05 Feb",
        "06 Feb",
        "07 Feb"
      ],
      labels: {
        style: {
          colors: ["#6B7280"],
          fontSize: "14px",
          fontWeight: 500
        }
      },
      axisBorder: {
        color: "#F3F4F6"
      },
      axisTicks: {
        color: "#F3F4F6"
      }
    },
    yaxis: {
      labels: {
        style: {
          colors: ["#6B7280"],
          fontSize: "14px",
          fontWeight: 500
        },
        formatter: function (value) {
          return "$" + value;
        }
      }
    },
    legend: {
      fontSize: "14px",
      fontWeight: 500,
      fontFamily: "Open Sans, sans-serif",
      labels: {
        colors: ["#6B7280"]
      },
      itemMargin: {
        horizontal: 10
      }
    },
    responsive: [
      {
        breakpoint: 1024,
        options: {
          xaxis: {
            labels: {
              show: false
            }
          }
        }
      }
    ]
  };

  const chart = new ApexCharts(
    document.getElementById("sales-overview"),
    options
  );
  chart.render();
}

if (document.getElementById("main-chart")) {
  const options = {
    series: [
      {
        name: "Users",
        data: [34, 45, 53, 38, 55, 32, 36]
      }
    ],
    labels: [
      "01 Feb",
      "02 Feb",
      "03 Feb",
      "04 Feb",
      "05 Feb",
      "06 Feb",
      "07 Feb"
    ],
    chart: {
      type: "bar",
      height: "420px",
      foreColor: "#4a576d",
      fontFamily: "Open Sans, sans-serif",
      toolbar: {
        show: false
      }
    },
    theme: {
      monochrome: {
        enabled: true,
        color: "#cb0c9f"
      }
    },
    plotOptions: {
      bar: {
        columnWidth: "15%",
        borderRadius: 6,
        colors: {
          backgroundBarColors: ["#4a576d", "#4a576d", "#4a576d", "#4a576d"],
          backgroundBarRadius: 6
        }
      },
      dataLabels: {
        hideOverflowingLabels: false
      }
    },
    xaxis: {
      floating: true,
      labels: {
        show: false
      },
      axisBorder: {
        show: false
      },
      axisTicks: {
        show: false
      }
    },
    tooltip: {
      shared: true,
      intersect: false,
      style: {
        fontSize: "14px",
        fontFamily: "Open Sans, sans-serif"
      }
    },
    states: {
      hover: {
        filter: {
          type: "darken",
          value: 0.8
        }
      }
    },
    fill: {
      opacity: 1
    },
    yaxis: {
      show: false
    },
    grid: {
      show: false
    },
    dataLabels: {
      enabled: false
    },
    legend: {
      show: false
    }
  };

  const chart = new ApexCharts(document.getElementById("main-chart"), options);
  chart.render();
}

if (document.getElementById("new-products-chart")) {
  const options = {
    colors: ["#0694a2", "#4E4187"],
    series: [
      {
        name: "Digital",
        color: "#0694a2",
        data: [
          { x: "01 Feb", y: 170 },
          { x: "02 Feb", y: 180 },
          { x: "03 Feb", y: 164 },
          { x: "04 Feb", y: 145 },
          { x: "05 Feb", y: 174 },
          { x: "06 Feb", y: 170 },
          { x: "07 Feb", y: 155 }
        ]
      },
      {
        name: "Goods",
        color: "#4E4187",
        data: [
          { x: "01 Feb", y: 120 },
          { x: "02 Feb", y: 134 },
          { x: "03 Feb", y: 167 },
          { x: "04 Feb", y: 179 },
          { x: "05 Feb", y: 145 },
          { x: "06 Feb", y: 182 },
          { x: "07 Feb", y: 143 }
        ]
      }
    ],
    chart: {
      type: "bar",
      height: "305px",
      fontFamily: "Open Sans, sans-serif",
      foreColor: "#4B5563",
      toolbar: {
        show: false
      }
    },
    plotOptions: {
      bar: {
        columnWidth: "50%",
        borderRadius: 2
      }
    },
    tooltip: {
      shared: true,
      intersect: false,
      style: {
        fontSize: "14px",
        fontFamily: "Open Sans, sans-serif"
      }
    },
    states: {
      hover: {
        filter: {
          type: "darken",
          value: 1
        }
      }
    },
    stroke: {
      show: true,
      width: 5,
      colors: ["transparent"]
    },
    grid: {
      show: false
    },
    dataLabels: {
      enabled: false
    },
    legend: {
      show: false
    },
    xaxis: {
      floating: true,
      labels: {
        show: false
      },
      axisBorder: {
        show: false
      },
      axisTicks: {
        show: false
      }
    },
    yaxis: {
      show: false
    },
    fill: {
      opacity: 1
    }
  };

  const chart = new ApexCharts(
    document.getElementById("new-products-chart"),
    options
  );
  chart.render();
}

if (document.getElementById("visitors-chart")) {
  const options = {
    series: [
      {
        name: "Visitors",
        data: [500, 590, 600, 520, 610, 550, 600]
      }
    ],
    labels: [
      "01 Feb",
      "02 Feb",
      "03 Feb",
      "04 Feb",
      "05 Feb",
      "06 Feb",
      "07 Feb"
    ],
    chart: {
      type: "area",
      height: "305px",
      fontFamily: "Open Sans, sans-serif",
      sparkline: {
        enabled: true
      },
      toolbar: {
        show: false
      }
    },
    fill: {
      type: "gradient",
      gradient: {
        shade: "light",
        shadeIntensity: 1
      }
    },
    plotOptions: {
      area: {
        fillTo: "end"
      }
    },
    theme: {
      monochrome: {
        enabled: true,
        color: "#0694a2"
      }
    },
    tooltip: {
      style: {
        fontSize: "14px",
        fontFamily: "Open Sans, sans-serif"
      }
    }
  };

  const chart = new ApexCharts(
    document.getElementById("visitors-chart"),
    options
  );
  chart.render();
}

if (document.getElementById("week-signups-chart")) {
  const options = {
    series: [
      {
        name: "Users",
        data: [34, 45, 53, 38, 55, 32, 36]
      }
    ],
    labels: [
      "01 Feb",
      "02 Feb",
      "03 Feb",
      "04 Feb",
      "05 Feb",
      "06 Feb",
      "07 Feb"
    ],
    chart: {
      type: "bar",
      height: "305px",
      foreColor: "#4B5563",
      fontFamily: "Open Sans, sans-serif",
      toolbar: {
        show: false
      }
    },
    theme: {
      monochrome: {
        enabled: true,
        color: "#0694a2"
      }
    },
    plotOptions: {
      bar: {
        columnWidth: "25%",
        borderRadius: 2,
        colors: {
          backgroundBarColors: ["#E5E7EB", "#E5E7EB", "#E5E7EB", "#E5E7EB"],
          backgroundBarRadius: 2
        }
      },
      dataLabels: {
        hideOverflowingLabels: false
      }
    },
    xaxis: {
      floating: true,
      labels: {
        show: false
      },
      axisBorder: {
        show: false
      },
      axisTicks: {
        show: false
      }
    },
    tooltip: {
      shared: true,
      intersect: false,
      style: {
        fontSize: "14px",
        fontFamily: "Open Sans, sans-serif"
      }
    },
    states: {
      hover: {
        filter: {
          type: "darken",
          value: 0.8
        }
      }
    },
    fill: {
      opacity: 1
    },
    yaxis: {
      show: false
    },
    grid: {
      show: false
    },
    dataLabels: {
      enabled: false
    },
    legend: {
      show: false
    }
  };

  const chart = new ApexCharts(
    document.getElementById("week-signups-chart"),
    options
  );
  chart.render();
}

if (document.getElementById("traffic-channels-chart")) {
  const options = {
    series: [30, 24, 18, 12, 9, 7],
    labels: ["Organic", "Referral", "Direct", "Social", "Other", "Email"],
    colors: ["#17c1e8", "#2152ff", "#129bba", "#cb0c9f", "#7928CA", "#141727"],
    chart: {
      type: "donut",
      height: 305,
      fontFamily: "Open Sans, sans-serif",
      toolbar: {
        show: false
      }
    },
    plotOptions: {
      pie: {
        donut: {
          size: "5%"
        }
      }
    },
    states: {
      hover: {
        filter: {
          type: "darken",
          value: 0.9
        }
      }
    },
    tooltip: {
      shared: true,
      followCursor: false,
      fillSeriesColor: false,
      inverseOrder: true,
      style: {
        fontSize: "14px",
        fontFamily: "Open Sans, sans-serif"
      },
      x: {
        show: true,
        formatter: function (_, { seriesIndex, w }) {
          const label = w.config.labels[seriesIndex];
          return label;
        }
      },
      y: {
        formatter: function (value) {
          return value + "%";
        }
      }
    },
    grid: {
      show: false
    },
    dataLabels: {
      enabled: false
    },
    legend: {
      show: false
    }
  };

  const chart = new ApexCharts(
    document.getElementById("traffic-channels-chart"),
    options
  );
  chart.render();
}
